modules/markup/html.go

```diff
@@ -72,7 +72,7 @@ var (
 const keywordClass = "issue-keyword"
 
 // regexp for full links to issues/pulls
-var issueFullPattern *regexp.Regexp
+var issueFullPattern = regexp.MustCompile(regexp.QuoteMeta(setting.AppURL) + `\w+/\w+/(?:issues|pulls)/((?:\w{1,10}-)?[1-9][0-9]*)([\?|#]\S+.(\S+)?)?\b`)
```

Comment:
setting.AppURL is not guaranteed to be set by the time that you first import modules/markup.
'
This will cause a dependency race that could be very very difficult to fix once it appears.

Reply:
aaaa yes how could I have missed that